tests/upgrade-tests/validation.py

```diff
@@ -18,6 +19,40 @@ def validate_status_messages(model):
             assert unit.data['workload-status']['message'] == message
 
 
+async def validate_snap_versions(model):
+    ''' Validate that the installed snap versions are consistent with channel
+    config on the charms.
+    '''
+    snaps_to_validate = {
+        'kubernetes-master': [
+            'kubectl',
+            'kube-apiserver',
+            'kube-controller-manager',
+            'kube-scheduler',
+            'cdk-addons',
+        ],
+        'kubernetes-worker': [
+            'kubectl',
+            'kubelet',
+            'kube-proxy',
+        ],
+    }
+    for app_name, snaps in snaps_to_validate.items():
```

Comment:
I would caution that we add some comment of the output snap list is expected to return so that when it breaks and our screen scrape parser misbehaves we know why. I ran into this with the layer-etcd snap refactoring.